ui/src/components/directory/viewer/Content.vue

```diff
@@ -125,7 +126,8 @@ async function init() {
 }
 
 async function getFavourites() {
-  const result = await EntityService.getPartialEntities(favourites.value, [RDFS.LABEL, RDF.TYPE]);
+  const favouriteList: string[] = currentUser.value ? await UserService.getUserFavourites(currentUser.value.id) : favourites ? favourites.value : "[]";
```

Comment:
Try to avoid using nested ternary statements, they can be difficult to read.  So instead of
```
const favouriteList: string[] = currentUser.value ? await UserService.getUserFavourites(currentUser.value.id) : favourites ? favourites.value : "[]";
```
split into 2
```
let favouriteList: string[];

if (currentUser.value) 
  favouriteList = await UserService.getUserFavourites(currentUser.value.id);
else
  favouriteList = favourites ? favourites.value : [];
```